Formula/newrelic-infra-agent.rb

```diff
@@ -0,0 +1,42 @@
+class NewrelicInfraAgent < Formula
+  desc "New Relic infrastructure agent"
+  homepage "https://github.com/newrelic/infrastructure-agent"
+  url "https://github.com/newrelic/infrastructure-agent/archive/refs/tags/1.20.3.tar.gz"
+  sha256 "3a29b19541e6f32fb87196a7833d08a9aed41b8ff62ef3a72326236d262cb033"
+  license "Apache-2.0"
+  head "https://github.com/newrelic/infrastructure-agent.git"
+
+  depends_on "go" => :build
```

Comment:
```suggestion
  depends_on "go" => :build
  depends_on arch: :x86_64
```
It seems you're blocked on ARM by a dependency, so we can do this. Let's try to fix the Linux build, though.